azure/services/scalesetvms/scalesetvms.go

```diff
@@ -155,7 +155,7 @@ func (s *Service) deleteVMSSFlexVM(ctx context.Context, resourceID string) error
 	resourceGroup := parsed.ResourceGroup
 	resourceName := strings.TrimPrefix(s.Scope.ProviderID(), azure.ProviderIDPrefix)
 	resourceNameSplits := strings.Split(resourceName, "/")
-	resourceName = resourceNameSplits[len(resourceNameSplits)-3] + "_" + resourceNameSplits[len(resourceNameSplits)-1]
+	resourceName = resourceNameSplits[len(resourceNameSplits)-1]
```

Comment:
/hold

I'll follow up with unit tests, code cleanup, etc. but for now just making sure this fixes the e2e test

Reply:
passed!